packages/main-api/src/controllers/v3/users/find_user.rs

```diff
@@ -0,0 +1,97 @@
+use crate::{
+    AppState, Error2,
+    models::user::{User, UserDetailResponse, UserMetadata, UserPhoneNumber},
+};
+use bdk::prelude::*;
+use dto::{
+    JsonSchema, aide,
+    by_axum::axum::{
+        Json,
+        extract::{Query, State},
+    },
+};
+use serde::Deserialize;
+
+use validator::Validate;
+
+#[derive(Debug, Clone, Deserialize, aide::OperationIo, JsonSchema, Validate)]
+#[serde(rename_all = "camelCase")]
+pub struct FindUserQueryParams {
+    #[schemars(description = "email address")]
+    #[validate(email)]
+    pub email: Option<String>,
+    #[schemars(description = "username")]
+    pub username: Option<String>,
+    #[schemars(description = "User's phone number")]
+    pub phone_number: Option<String>,
+}
```

Comment:
```suggestion
pub struct FindUserQueryParams {
    #[schemars(description = "email address")]
    pub r#type: FindUserQueryType,
    #[schemars(description = "username")]
    pub value: String,
}

pub enum FindUserQueryType {
      Email,
      Username,
      PhoneNumber,
}
```
이런식으로 구성하는게 좋을거 같아요. 위의 처럼 하면 사람들 입장에서는 여러개로 한번에 찾을수 있다고 생각할 수 있거든요.